frontend/src/components/LoginModal.js

```diff
@@ -0,0 +1,247 @@
+                if (data.code === 1) {
+                    const cookies = new Cookies();
+                    cookies.set('token', data.token);
+                    this.setState({ redirect: '/' });
+                } else {
+                    console.log(data);
+                }
+            });
+        } else if (!showRegisterForm) {
+            axios({
+                url: `http://${process.env.REACT_APP_IP}/api/auth/login`,
+                method: 'post',
+                data: {
+                    username,
+                    password,
+                },
+            })
+            .then(({ data }) => {
+                if (data.code === 1) {
+                    const { token } = data;
+                    const cookies = new Cookies();
+                    cookies.set('token', token);
+                    window.location.reload();
+                } else {
+                    console.log(data);
+                }
+            })
+            .catch(error => {
+                const { errors } = this.state;
+                errors.loginError = 'Incorrect username or password!';
+                this.setState({
+                    errors,
+                    isWorking: false,
+                });
+            });
+        }
+    }
+
+    showRegisterForm = () => {
+        const showRegisterForm = this.state.showRegisterForm;
+        this.setState({ 
+            ...this.initializeForm(),
+            showRegisterForm: !showRegisterForm,
+        });
+    }
+
+    onClose = () => {
+        this.props.onClose();
+        setTimeout(() => {
+            this.setState(this.initializeForm());
+        }, 1000);
+    }
+
+    render(){
+        const { showRegisterForm, isWorking, redirect } = this.state; 
+        const { username, password, verify } = this.state.fields;
+        const { loginError, userError, passError, verifyError } = this.state.errors;
+        
+        const errorIndicator = {
+            borderBottom: '2px solid #ff0000',
+        };
+
+        let formValid;
+        if (showRegisterForm) {
+            formValid = username.length > 0 && password.length > 0 && verify.length > 0 && userError.concat(passError, verifyError).length === 0;
+        } else {
+            formValid = username.length > 0 && password.length > 0 && userError.concat(passError).length === 0;
+        }
+
+        return (
+            <Modal 
+                open={this.props.open} 
+                showCloseIcon={true}
+                closeOnEsc={true}
+                onClose={this.onClose}
+                classNames={{ overlay: 'modal-overlay', modal: 'modal-container' }} 
+                little>
+                {
+                    redirect && <Redirect to={redirect}/>
+                }
+                <h1 className="modal-header">{ showRegisterForm ? 'Sign up for CaptionThis' : 'Log in to your account' }</h1>
+                <form className="modal-login-form" onSubmit={this.onSubmit}>
```

Comment:
This modal is for login and register, maybe another name like modal-authentication-form would be more significant. This applies for the rest of the components in here.